src/main/java/io/icker/factions/ui/MemberGui.java

```diff
@@ -0,0 +1,144 @@
+package io.icker.factions.ui;
+
+import com.mojang.authlib.GameProfile;
+import eu.pb4.sgui.api.ClickType;
+import eu.pb4.sgui.api.elements.GuiElementBuilder;
+import eu.pb4.sgui.api.gui.SimpleGui;
+import io.icker.factions.api.persistents.Faction;
+import io.icker.factions.api.persistents.User;
+import io.icker.factions.command.RankCommand;
+import io.icker.factions.util.Command;
+import io.icker.factions.util.GuiInteract;
+import io.icker.factions.util.Message;
+import net.minecraft.component.DataComponentTypes;
+import net.minecraft.component.type.ProfileComponent;
+import net.minecraft.item.Items;
+import net.minecraft.screen.ScreenHandlerType;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Style;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.UserCache;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MemberGui extends PagedGui {
+    Faction faction;
+    int size;
+    UserCache cache;
+    User user;
+
+    List<User> members;
+
+    public MemberGui(ServerPlayerEntity player, Faction faction, @Nullable Runnable closeCallback) {
+        super(player, closeCallback);
+        this.faction = faction;
+        this.cache = player.getServer().getUserCache();
+        this.user = User.get(player.getUuid());
+
+        this.members = new ArrayList<>(faction.getUsers());
+        this.members.remove(user);
+        this.members.addFirst(user);
+        this.size = members.size();
+
+        this.setTitle(Text.literal(faction.getColor() + faction.getName()).append(Text.literal(" members list (" + size + ")")));
+        this.updateDisplay();
+        this.open();
+    }
+
+    @Override
+    protected int getPageAmount() {
+        return this.size / PAGE_SIZE;
+    }
+
+    @Override
+    protected DisplayElement getElement(int id) {
+        if (this.size > id) {
+            var targetUser = this.members.get(id);
+            GameProfile profile = cache.getByUuid(targetUser.getID()).orElseThrow();
```

Comment:
Instead of throwing an error, this should at least just show an empty slot or a head with a question mark or something similar.